File: app/src/main/java/com/sc/clgg/util/ConfigUtil.java
//package com.sc.clgg.util;
//
//import android.content.Context;
//import android.content.SharedPreferences;
//import android.content.SharedPreferences.Editor;
//
//import com.sc.clgg.application.App;
//import com.sc.clgg.bean.UserInfoBean;
//import com.sc.clgg.config.Constants;
//
//public class ConfigUtil {
//    private SharedPreferences sp;
//    private Editor editor;
//
//    public ConfigUtil() {
//        super();
//        sp = App.getInstance().getSharedPreferences("user_info2", Context.MODE_PRIVATE);
//        editor = sp.edit();
//    }
//
//    public ConfigUtil(Context context) {
//        super();
//        sp = context.getSharedPreferences("user_info2", Context.MODE_PRIVATE);
//        editor = sp.edit();
//    }
//
//    public String getUserType() {
//        return sp.getString(Constants.USERTYPE, "");
//    }
//
//    public Editor getEditor() {
//        return editor;
//    }
//
//    public void setEditor(Editor editor) {
//        this.editor = editor;
//    }
//
//    public String getEmail() {
//        return sp.getString(Constants.EMALI, "");
//    }
//
//    public void setEmail(String email) {
//        editor.putString(Constants.EMALI, email);
//        editor.apply();
//    }
//
//    public String getGender() {
//        return sp.getString(Constants.GENDER, "");
//    }
//
//    public void setGender(String gender) {
//        editor.putString(Constants.GENDER, gender);
//        editor.apply();
//    }
//
//
//    public String getMobile() {
//        return sp.getString(Constants.MOBILE, "");
//    }
//
//    public void setMobile(String mobile) {
//        editor.putString(Constants.MOBILE, mobile);
//        editor.apply();
//    }
//
//    public String getPassword() {
//        return sp.getString(Constants.PASSWORD, "");
//    }
//
//    public void setPassword(String password) {
//        editor.putString(Constants.PASSWORD, password);
//        editor.apply();
//    }
//
//    public SharedPreferences getSp() {
//        return sp;
//    }
//
//    public void setSp(SharedPreferences sp) {
//        this.sp = sp;
//    }
//
//    public String getUserid() {
//        return sp.getString(Constants.USERID, "");
//    }
//
//    public void setUserid(String userid) {
//        editor.putString(Constants.USERID, userid);
//        editor.apply();
//    }
//
//    public String getUsername() {
//        return sp.getString(Constants.USERNAME, "");
//    }
//
//    public void setUsername(String username) {
//        editor.putString(Constants.USERNAME, username);
//        editor.apply();
//    }
//
//    public boolean setUserInfo(UserInfoBean bean) {
//        if (null != bean) {
//            editor.putString(Constants.USERID, bean.getUserCode());
//            editor.putString(Constants.USERNAME, bean.getUserName());
//            editor.putString(Constants.MOBILE, bean.getPersonalPhone());
//            editor.putString(Constants.USERTYPE, bean.getUserType());
//            editor.commit();
//            return true;
//        } else {
//            return false;
//        }
//    }
//
//    public void reset() {
//        setUserid("");
//        setUsername("");
//        setMobile("");
//        setPassword("");
//    }
//
//}

File: app/src/main/java/com/sc/clgg/util/ConfigUtil.kt
package com.sc.clgg.util

import android.content.Context
import android.content.Intent
import android.content.SharedPreferences
import android.text.TextUtils
import androidx.core.content.edit
import com.sc.clgg.activity.login.LoginRegisterActivity
import com.sc.clgg.application.App
import com.sc.clgg.bean.UserInfoBean

/**
 * 保存用户信息
 */
class ConfigUtil {
    private var sharedPreferences: SharedPreferences? = null

    init {
        sharedPreferences = App.getInstance().getSharedPreferences("user_info2", Context.MODE_PRIVATE)
    }

    var icon: String?
        get() = sharedPreferences?.getString("icon", "")
        set(icon) {
            sharedPreferences?.edit { putString("icon", icon) }
        }

    var mobile: String?
        get() = sharedPreferences?.getString("mobile", "")
        set(mobile) {
            sharedPreferences?.edit { putString("mobile", mobile) }
        }

    var userid: String
        get() = sharedPreferences?.getString("userid", "") ?: ""
        set(userid) {
            sharedPreferences?.edit { putString("userid", userid) }
        }

    var username: String?
        get() = sharedPreferences?.getString("username", "")
        set(username) {
            sharedPreferences?.edit { putString("username", username) }
        }
    /**
     * 无车承运人账号
     */
    var account: String?
        get() = sharedPreferences?.getString("account", "")
        set(account) {
            sharedPreferences?.edit { putString("account", account) }
        }

    var realName: String?
        get() = sharedPreferences?.getString("realName", "")
        set(username) {
            sharedPreferences?.edit { putString("realName", username) }
        }
    var nickName: String?
        get() = sharedPreferences?.getString("nickName", "")
        set(nickName) {
            sharedPreferences?.edit { putString("nickName", nickName) }
        }
    var password: String?
        get() = sharedPreferences?.getString("password", "")
        set(password) {
            sharedPreferences?.edit { putString("password", password) }
        }

    /**
     * 更新应用中存储的用户信息
     */
    fun setUserInfo(bean: UserInfoBean?) {
        sharedPreferences?.edit {
            putString("userid", bean?.userCode)
            putString("username", bean?.userName)
            putString("mobile", bean?.personalPhone)
            putString("password", bean?.password)
            putString("account", bean?.userName)
        }
    }

    /**
     * 删除应用中存储的用户信息
     */
    fun clear() {
        sharedPreferences?.edit {
            clear()
        }
    }

    /**
     * 是否已经登录
     */
    fun loggedIn(context: Context?): Boolean {
        return if (TextUtils.isEmpty(userid)) {
            context?.startActivity(Intent(context, LoginRegisterActivity::class.java))
            false
        } else {
            true
        }
    }
}
